internal/backend/utils.go

```diff
@@ -10,24 +10,11 @@ import (
 
 // LoadAll reads all data stored in the backend for the handle.
 func LoadAll(ctx context.Context, be restic.Backend, h restic.Handle) (buf []byte, err error) {
-	rd, err := be.Load(ctx, h, 0, 0)
-	if err != nil {
-		return nil, err
-	}
-
-	defer func() {
-		_, e := io.Copy(ioutil.Discard, rd)
-		if err == nil {
-			err = e
-		}
-
-		e = rd.Close()
-		if err == nil {
-			err = e
-		}
-	}()
-
-	return ioutil.ReadAll(rd)
+	err = be.Load(ctx, h, 0, 0, func(rd io.Reader) error {
+		buf, err = ioutil.ReadAll(rd)
```

Comment:
Hm, I'm not sure if we should use the same `err` variable within the func and outside. That may lead to problems when the func is called repeatedly. What are your thoughts?

What about this:
```go
err = be.Load(ctx, h, 0, 0, func(rd io.Reader) (ierr error) {
￼		buf, ierr = ioutil.ReadAll(rd)
        return ierr
￼￼})
```

This happens a lot further down in the commit.